streams/src/test/java/org/apache/kafka/streams/processor/internals/StateManagerUtilTest.java

```diff
@@ -249,32 +247,35 @@ public void testCloseStateManagerOnlyThrowsFirstExceptionWhenClean() throws IOEx
     }
 
     @Test
-    public void testCloseStateManagerDirtyShallSwallowException() throws IOException {
-        final LogCaptureAppender appender = LogCaptureAppender.createAndRegister();
-
+    public void testCloseStateManagerThrowsExceptionWhenDirty() throws IOException {
```

Comment:
We actually were already throwing some `ProcessorStateExceptions` up through  close even when unclean, which I think was the cause of a bug we resolved a few weeks ago. Now we just make no assumptions about whether this will throw or not, and catch any exceptions in `StreamTask` / `StandbyTask` if it's a dirty close.